src/Bicep.Core/Semantics/SemanticModel.cs

```diff
@@ -188,10 +189,33 @@ public IEnumerable<IDiagnostic> GetAllDiagnostics()
 
         private IEnumerable<IDiagnostic> AssembleDiagnostics()
         {
-            return GetParseDiagnostics()
-            .Concat(GetSemanticDiagnostics())
-            .Concat(GetAnalyzerDiagnostics())
-            .OrderBy(diag => diag.Span.Position);
+            var diagnostics = GetParseDiagnostics()
+                .Concat(GetSemanticDiagnostics())
+                .Concat(GetAnalyzerDiagnostics())
+                .OrderBy(diag => diag.Span.Position);
+            List<IDiagnostic> updatedDiagnostics = new List<IDiagnostic>();
```

Comment:
[nit] Can we name this `filteredDiagnostics`? We're also trying to use `var` by default where the context is clear (e.g. `var filteredDiagnostics = new List<IDiagnostic>();`)

Reply:
Updated as part of this commit:
https://github.com/Azure/bicep/commit/7f012613ae1aad9fd2e8101ac82b73b362b8686b

Thanks!